keras_spiking/regularizers.py

```diff
@@ -65,137 +139,127 @@ class L2(L1L2):
+    In the following example, we use `.Percentile` to ensure the neuron activities
     (a.k.a., firing rates) fall in the desired range of 5-10 Hz when computing the
     product of two inputs.
 
     .. testcode::
 
-       train_x = np.random.uniform(-1, 1, size=(1024 * 100, 2))
-       train_y = train_x[:, :1] * train_x[:, 1:]
-       test_x = np.random.uniform(-1, 1, size=(128, 2))
-       test_y = test_x[:, :1] * test_x[:, 1:]
-
-       # train using one timestep, since spiking_aware_training=False
-       train_seq = train_x[:, None]
-
-       # test using 10 timesteps
-       n_steps = 10
-       test_seq = np.tile(test_x[:, None], (1, n_steps, 1))
-
-       model = tf.keras.Sequential(
-           [
-               tf.keras.layers.TimeDistributed(
-                   tf.keras.layers.Dense(50), input_shape=(None, 2)
-               ),
-               keras_spiking.SpikingActivation(
-                   "relu",
-                   dt=1,
-                   spiking_aware_training=False,
-                   activity_regularizer=keras_spiking.PercentileRange(
-                       scale=0.01, minimum=5, maximum=10, penalty="l2"
-                   ),
-               ),
-               tf.keras.layers.GlobalAveragePooling1D(),
-               tf.keras.layers.Dense(1),
-           ]
-       )
-       model.compile(
-           optimizer="rmsprop",
-           loss="mse",
-       )
-       model.fit(train_seq, train_y, epochs=5)
-
-       # create functional model to return spiking layer outputs too
-       func_model = tf.keras.Model(model.input, [model.output, model.layers[1].output])
-       outputs, spikes = func_model.predict(test_seq)
-
-       rates = spikes.mean(axis=1)  # estimate rates by averaging over time
-       max_rates = rates.max(axis=0)
-       print("Max rates: %s, %s" % (max_rates.mean(), max_rates.std()))
-
-       error = np.mean(np.abs(outputs - test_y))
-       print("Error: %s" % (error,))
+        train_x = np.random.uniform(-1, 1, size=(1024 * 100, 2))
+        train_y = train_x[:, :1] * train_x[:, 1:]
+        test_x = np.random.uniform(-1, 1, size=(128, 2))
+        test_y = test_x[:, :1] * test_x[:, 1:]
+
+        # train using one timestep, to speed things up
+        train_seq = train_x[:, None]
+
+        # test using 10 timesteps
+        n_steps = 10
+        test_seq = np.tile(test_x[:, None], (1, n_steps, 1))
+
+        inp = x = tf.keras.Input((None, 2))
+        x = tf.keras.layers.TimeDistributed(tf.keras.layers.Dense(50))(x)
+        x = spikes = keras_spiking.SpikingActivation(
+            "relu",
+            dt=1,
+            activity_regularizer=keras_spiking.regularizers.Percentile(target=(5, 10)),
+        )(x)
+        x = tf.keras.layers.GlobalAveragePooling1D()(x)
+        x = tf.keras.layers.Dense(1)(x)
+
+        model = tf.keras.Model(inp, (x, spikes))
+
+        model.compile(
+            optimizer="rmsprop", loss={"dense_1": "mse"}, metrics={"dense_1": "mae"}
```

Comment:
Why do we need to specify "dense_1" here? Is that because the model outputs "spikes" too, and without specifying that, it would also apply "mse" to the spikes? If that's the case, I should definitely re-examine my development of this example, because that could have been causing a lot of my problems.

Also, we should really highlight that for users (with a comment at least), since that seems like an easy/common pitfall.